modules/queue/unique_queue_disk_channel.go

```diff
@@ -188,7 +188,7 @@ func (q *PersistableChannelUniqueQueue) Run(atShutdown, atTerminate func(func())
 		go q.internal.Run(func(_ func()) {}, func(_ func()) {})
 		go func() {
 			_ = q.internal.Flush(0)
-			log.Debug("LevelUniqueQueue: %s flushed so shutting down", q.internal.(*LevelQueue).Name())
```

Comment:
I want to know if  line 197 also may panic?

Reply:
nope it can't

Reply:
But it also `q.internal.(*LevelUniqueQueue)` when `luq, ok := q.internal.(*LevelUniqueQueue); !ok`

Reply:
it's defined as LevelUniqueQueue.

Reply:
so maybe `if q.internal.(*LevelUniqueQueue).ByteFIFOUniqueQueue.byteFIFO.Len(luq.shutdownCtx) != 0` will not make confusion.